packages/@uppy/webdav/src/WebdavOauth.jsx

```diff
@@ -0,0 +1,100 @@
+import { h } from 'preact'
+import { useCallback, useState } from 'preact/hooks'
+
+import { UIPlugin } from '@uppy/core'
+import { Provider } from '@uppy/companion-client'
+import { ProviderViews } from '@uppy/provider-views'
+
+import packageJson from '../package.json'
+import locale from './locale.js'
+
+class WebdavOauthProvider extends Provider {
+  // eslint-disable-next-line class-methods-use-this
+  authQuery ({ authFormData }) {
+    const { webdavUrl } = authFormData
+
+    return {
+      subdomain: new URL(webdavUrl).host || webdavUrl,
+    }
+  }
+}
+
+const AuthForm = ({ loading, i18n, onAuth }) => {
+  const [webdavUrl, setWebdavUrl] = useState('')
+
+  const onSubmit = useCallback((e) => {
+    e.preventDefault()
+    onAuth({ webdavUrl })
+  }, [onAuth, webdavUrl])
+
+  return (
+    <form onSubmit={onSubmit}>
+      <label htmlFor="uppy-Provider-publicLinkURL">
+        <span>{i18n('publicLinkURLLabel')}</span>
+        <input id="uppy-Provider-publicLinkURL" name="webdavUrl" type="text" value={webdavUrl} onChange={(e) => setWebdavUrl(e.target.value)} disabled={loading} />
+      </label>
+      <span>{i18n('publicLinkURLDescription')}</span>
+
+      <button style={{ display: 'block' }} disabled={loading} type="submit">Submit</button>
+    </form>
```

Comment:
For the Oauth variant `publicLinkURL` in specified names probably doesn't make sense, we rather want to enter just a "server url" I think